presto-hive/src/main/java/io/prestosql/plugin/hive/HivePageSourceProvider.java

```diff
@@ -678,6 +691,36 @@ public int getBucketToKeep()
         }
     }
 
+    private static Optional<BucketValidator> createBucketValidator(Path path, Optional<BucketValidation> bucketValidation, OptionalInt bucketNumber, List<ColumnMapping> columnMappings)
+    {
+        return bucketValidation.flatMap(validation -> {
+            Map<Integer, ColumnMapping> baseHiveColumnToBlockIndex = columnMappings.stream()
+                    .filter(mapping -> mapping.getHiveColumnHandle().isBaseColumn())
+                    .collect(toImmutableMap(mapping -> mapping.getHiveColumnHandle().getBaseHiveColumnIndex(), identity()));
+
+            int[] bucketColumnIndices = new int[validation.getBucketColumns().size()];
+
+            List<TypeInfo> bucketColumnTypes = new ArrayList<>();
+            for (int i = 0; i < validation.getBucketColumns().size(); i++) {
+                HiveColumnHandle column = validation.getBucketColumns().get(i);
+                ColumnMapping mapping = baseHiveColumnToBlockIndex.get(column.getBaseHiveColumnIndex());
+                if (mapping == null) {
```

Comment:
when is that possible? Can you add a comment?

Reply:
This happens when we are not reading a bucket column (in which case we don't need to bother with validation since it can't affect the query results). I'll add a comment.